### ISSUE TITLE
Remove use of Proxy for events in development

### ISSUE THREAD

**Issue by gaearon** [facebook/react#12171]
I think maybe we should revert https://github.com/facebook/react/pull/5947.

People already think `proxyEvent` is some kind of an API: https://github.com/facebook/react/issues/12169.

It's also annoying to view in the debugger because none of the properties show up.

Instead, we could seal the event object or something like that.

**Comment by koba04** [comment]
`Object.seal` can seal the event object but in a non strict environment, adding extra properties doesn't warn anything, just ignoring it.
It makes debugging so hard.

(In strict mode, we can get an error something like this.)

> Uncaught TypeError: Cannot add property foo, object is not extensible

It's a trade off :)

**Comment by gaearon** [comment]
I think it's safe to assume most React users run in strict mode by now.

**Comment by jquense** [comment]
You'd still run into the initial concern in https://github.com/facebook/react/issues/5853 that Sealing would prevent extension after `persist()`, granted that's probably not a great pattern but i bet plently of folks will show up to say some integral abstraction is built on that "feature" :P

**Comment by gaearon** [comment]
Can we rewrite the check to just verify the existing fields are not used when we take it from the pool? I just want to get rid of the proxy there. It's annoying.

**Comment by koba04** [comment]
I'll work on to verify that the change makes sense.

**Comment by Kovensky** [comment]
If you are using an ES Proxy and the properties don't show up it sounds like a debugger bug.

**Comment by gaearon** [comment]
I don't think so. Trying to enumerate them is potentially side-effectful, it wouldn't be safe for the debugger to do.

**Comment by Kovensky** [comment]
Only if there is a `get` or `ownKeys` trap; but that may be too specific for a debugger 🤔

**Comment by gaearon** [comment]
Fixed by https://github.com/facebook/react/pulls.
